tests/functional/context_methods/test_builtin_functions.py

```diff
@@ -55,6 +63,46 @@ def test_builtin_zip_function(self, project):
         assert f"zip_result: {expected_zip}" in log_output
         assert f"zip_strict_result: {expected_zip}" in log_output
 
+    def test_builtin_invocation_args_dict_function(self, project):
+        _, log_output = run_dbt_and_capture(
+            [
+                "--debug",
+                "--log-format=json",
+                "run-operation",
+                "validate_invocation",
+                "--args",
+                "{my_variable: test_variable}",
+            ]
+        )
+
+        parsed_logs = []
+        for line in log_output.split("\n"):
+            try:
+                log = json.loads(line)
+            except ValueError:
+                continue
+
+            parsed_logs.append(log)
+
+        # Value of msg is a string
+        result = next(
+            (
+                item
+                for item in parsed_logs
+                if "invocation_result" in item["data"].get("msg", "msg")
+            ),
+            False,
+        )
+
+        assert result
+
+        # Result is checked in two parts because profiles_dir is unique each test run
+        expected = "invocation_result: {'debug': True, 'log_format': 'json', 'write_json': True, 'use_colors': True, 'printer_width': 80, 'version_check': True, 'partial_parse': True, 'static_parser': True, 'profiles_dir': "
+        assert expected in str(result)
+
+        expected = "'send_anonymous_usage_stats': False, 'event_buffer_size': 100000, 'quiet': False, 'no_print': False, 'macro': 'validate_invocation', 'args': '{my_variable: test_variable}', 'which': 'run-operation', 'rpc_method': 'run-operation', 'indirect_selection': 'eager'}"
+        assert expected in str(result)
```

Comment:
I opted to add `--log-format=json` to the dbt method and parse the json logs then implement the asserts on this within the test rather than having it in the macro jinja.

Based on similar logic I found in [test_query_comments](https://github.com/dbt-labs/dbt-core/blob/main/test/integration/051_query_comments_tests/test_query_comments.py#L55-L72)